groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -2986,6 +2978,31 @@ class optional {
         return UnspecifiedBoolUtil::makeValue(has_value());
     }
 #endif  // BSLS_COMPILERFEATURES_SUPPORT_OPERATOR_EXPLICIT
+
+  protected:
+
+#ifndef BDE_OMIT_INTERNAL_DEPRECATED
+    // PROTECTED MANIPULATORS
+    TYPE& dereferenceRaw();
+        // Return a reference providing modifiable access to the underlying
+        // 'TYPE' object.  The behavior is undefined if the 'optional' object
+        // is disengaged.  Note that this function is only intended to be
+        // called by 'bdlb::NullableValue::value' during transition of its
+        // implementation to use 'bsl::optional.  Because 'NullableValue' does
+        // not have ref-qualified version of 'value()', it makes no sense to
+        // provide ref-qualified version of 'dereferenceRaw()'.
```

Comment:
```suggestion
        // implementation to use 'bsl::optional.  Note that ref-qualified versions
        // of 'value()' are not provided because 'NullableValue' does not require them.
```